server/expression/in_tuple.go

```diff
@@ -177,10 +199,6 @@ func (it *InTuple) WithChildren(children ...sql.Expression) (sql.Expression, err
 			if compFuncs[i] == nil {
 				return nil, fmt.Errorf("operator does not exist: %s = %s", leftType.String(), rightType.String())
 			}
-			if compFuncs[i].Type().(pgtypes.DoltgresType).BaseID() != pgtypes.DoltgresTypeBaseID_Bool {
-				// This should never happen, but this is just to be safe
-				return nil, fmt.Errorf("%T: found equality comparison that does not return a bool", it)
-			}
```

Comment:
Any reason to remove this? It's a sanity check, since I think it's _technically_ allowable to create a comparison function that doesn't return `bool`. I'm pretty sure you'll blow up Postgres by doing so, but we can at least fail gracefully.

Reply:
This happens with a null literal (return type of comparison is `UNKNOWN`). I'll see if I can re-enable this in the other cases.